### PR TITLE
Add a state property to determine resume behavior

### PR DESCRIPTION
**Pull request by rasmustautsglia**
**What was solved?**

If video call and screensharing were not active in Call Visualizer session
calling resume() resulted in screen sharing view being opened. An unexpected behavior. To make it much more readable, state property was introduced.

In addition, some code was moved around to make the class more readable.

**Release notes:**

 - [ ] Feature
 - [X] Ignore
 - [ ] Release notes (Is it clear from the description here?)
 - [ ] Migration guide (If changes are needed for integrator already using the SDK - what needs to be communicated? Add underneath please)

**Additional info:**

- [ ] Is the feature sufficiently tested? All tests fixed? Necessary unit, acceptance, snapshots added? Check that at least new public classes & methods are covered with unit tests
 - [ ] Did you add logging beneficial for troubleshooting of customer issues?
 - [ ] **Did you add new logging?** We would like the logging between platforms to be similar. Refer to **Logging from iOS SDKs** → **Things to consider for newly added logs** in Confluence for more information.

**Screenshots:**

### CODE REVIEW

**Review by ykyivskyi-gl — APPROVED** [review]

File: GliaWidgets/Sources/CallVisualizer/Coordinator/CallVisualizer.Coordinator.swift [hunk 1]
```diff
@@ -153,16 +50,126 @@ extension CallVisualizer {
...
+            handleEngagementRequestOutcomeNil(answer: answer)
+            return
+        }
+        handleEngagementRequestOutcomeTimeout(answer: answer)
+    }
+
+    func handleEngagementRequestOutcomeTimeout(answer: Command<Bool>) {
+        environment.alertManager.dismissCurrentAlert()
+        environment.gcd.mainQueue.asyncAfterDeadline(.now() + 0.5) {
+            // Will be swapped for localized string
```
Comment (igorkravchenko) [inline GliaWidgets/Sources/CallVisualizer/Coordinator/CallVisualizer.Coordinator.swift:115]: What about this? Is there a ticket to provide localised strings?

**Review by igorkravchenko — COMMENTED** [review]
Left one comment about localised string, which is not related probably to these changes, but still it has to be addressed sooner or later.

File: GliaWidgets/Sources/CallVisualizer/Coordinator/CallVisualizer.Coordinator.swift (hunk 1 above)
Comment (rasmustautsglia) [reply GliaWidgets/Sources/CallVisualizer/Coordinator/CallVisualizer.Coordinator.swift]: Ticket for it has been created and comment updated [ad3bf2e](https://github.com/salemove/ios-sdk-widgets/pull/1161/commits/ad3bf2e94dbeec8044b15a5b2a7be28472e02139)

**Review by igorkravchenko — APPROVED** [review]

**Comment by rasmustautsglia** [comment]
!squash

**Pull request merged**